pkcs11 support [12826] (#296)
eProsima/Fast-DDS-docs: docs/installation/sources/sources_linux.rst

   sudo apt install libssl-dev

.. _libp11_sl:

Libp11 and SoftHSM libraries
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Libp11 provides PKCS#11 support for OpenSSL. This is an optional dependency,
that is needed only when *eprosima Fast DDS* is used with security and PKCS#11 URIs.

Install libp11_ using the package manager of the appropriate Linux distribution.
For example, on Ubuntu use the command:

.. code-block:: bash

   sudo apt install libp11-dev libengine-pkcs11-openssl

SoftHSM is a software implementation of an HSM (Hardware Security Module).
If *eProsima Fast DDS* tests are activated and *libp11* is installed

> on the system, SoftHSM is additioanlly required to run tests of PKCS#11 features.

Reviewer: ```suggestion
on the system, SoftHSM is additionally required to run tests of PKCS#11 features.
```

To install SoftHSM_, download the repository and follow the installation instructions:

.. code-block:: bash

   git clone https://github.com/opendnssec/SoftHSMv2.git
   cd SoftHSMv2 && ./autogen.sh
   ./configure --with-crypto-backend=openssl --disable-non-paged-memory

>    make && sudo make install

Reviewer: I have installed it directly from the apt repository when I run the tests locally (in fact the configuration you mention to make openSSL aware of the engine, I did not have to do it. It seems it is done directly in Ubuntu because I could run the tests without any trouble. The only issue was that I had to run the tests with root privileges because otherwise softHSM2 complains):

```
sudo apt install softhsm2
```

that is needed only when *eprosima Fast DDS* is used with security and PKCS#11 URIs.

Install libp11_ using the package manager of the appropriate Linux distribution.
For example, on Ubuntu use the command:

.. code-block:: bash

   sudo apt install libp11-dev libengine-pkcs11-openssl

SoftHSM is a software implementation of an HSM (Hardware Security Module).
If *eProsima Fast DDS* tests are activated and *libp11* is installed
on the system, SoftHSM is additionally required to run tests of PKCS#11 features.

Install SoftHSM_ using the package manager of the appropriate Linux distribution.
For example, on Ubuntu use the command:

.. code-block:: bash

   sudo apt install softhsm2

OpenSSL access HSM and other hardware devices through its engine functionality.
In order to set up a new engine the OpenSSL configuration files (usually `/etc/ssl/openssl.cnf`)
must be updated specifying the libp11_ and hardware module (here SoftHSM_) dynamic libraries
location.

This configuration step can be avoided using p11kit_ which allows OpenSSL to find PKCS#11
devices on runtime without static configuration. This kit is often available through

> the distro package manager. On Ubuntu **apt**, for example: 

Reviewer: ```suggestion
the Linux distribution package manager. On Ubuntu, for example:
```

distro is not a valid word according with the spelling test.

`apt` is not mentioned in the rest of the file.

.. code-block:: bash

   sudo apt install libengine-pkcs11-openssl

Once installed, to check p11kit_ is able to find the SoftHSM_ module use:

.. code-block:: bash

   p11-kit list-modules

In order to check if OpenSSL is able to access PKCS#11 engine use:

.. code-block:: bash

>     openssl engine pkcs11 -t  

Reviewer: ```suggestion
    openssl engine pkcs11 -t
```
